src/Windows/Avalonia.Direct2D1/Direct2D1Platform.cs

```diff
@@ -31,15 +29,16 @@ public class Direct2D1Platform : IPlatformRenderInterface
     {
         private static readonly Direct2D1Platform s_instance = new Direct2D1Platform();
 
-        private static SharpDX.Direct2D1.Factory s_d2D1Factory;
+        public static SharpDX.Direct2D1.Factory1 Direct2D1Factory { get; private set; }
```

Comment:
Does set it to `internal` is enough?

Reply:
I do not think we should use it out of the framework

Reply:
I think there are scenarios where you need to access those. We even have a demo that needs access.

Reply:
@Gillibald We can use `[assembly: InternalsVisibleTo("xxdemo")]` to make the demo can use this property. The more API be public the more compatibility problem.

Reply:
The demo shows that it needs to be public. Otherwise, no one could write such an application with Avalonia. I could be wrong. Before I have made this change to Direct2D1Platform everything was public threw the AvaloniaLocator. Let us gather some opinions about that to make sure we don't make any mistake.